ufl/argument.py

```diff
@@ -44,6 +47,8 @@ def __init__(self, function_space, number, part=None):
             element = function_space
             domain = default_domain(element.cell())
             function_space = FunctionSpace(domain, element)
+            warnings.warn("The use of FiniteElement will be deprecated by December 2023. "
+                          "Please, use FunctionSpace instead", DeprecationWarning)
```

Comment:
Minor: use `FutureWarning`, not `DeprecationWarning`.

Rationale: The python interpreter does not present `DeprecationWarning`s to users by default (but does do so with `FutureWarning`s), see https://docs.python.org/3/library/warnings.html#warning-categories for details.

e.g.

`foo.py`
```python
import warnings


class X:
    def __init__(self):
        warnings.warn("This is a deprecation warning", DeprecationWarning)
        warnings.warn("This is a future warning", FutureWarning)
```
```
$ python -c "import foo; foo.X()"
/home/wence/Documents/src/doodles/python/foo.py:7: FutureWarning: This is a future warning
```
(Note how the user doesn't see the deprecation warning).